### PR TITLE
Add Intel Gaudi Accelerators to Ray Resources list

### PR DESCRIPTION
**Pull request by jerome-habana**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/util/accelerators/__init__.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
     GOOGLE_TPU_V2,
     GOOGLE_TPU_V3,
     GOOGLE_TPU_V4,
+    GAUDI,
```
Comment (ericl) [inline python/ray/util/accelerators/__init__.py:13]:
```suggestion
    INTEL_GAUDI,
```

File: python/ray/train/torch/config.py [hunk 2]
```diff
@@ -14,6 +14,10 @@
 import torch
 import torch.distributed as dist
 
+from ray._private.ray_constants import HPU_PACKAGE_AVAILABLE
```
Comment (jjyao) [inline python/ray/train/torch/config.py:17]: For Train related changes, could you do it in a separate PR?

**Review by jjyao — COMMENTED** [review]
lg

File: python/ray/train/torch/config.py (hunk 2 above)
Comment (jerome-habana) [reply python/ray/train/torch/config.py:17]: Sure. Do you think, there will be more changes for train other than this to support all Ray functionalities?

**Comment by jjyao** [comment]
Is this ready for review?

**Comment by anyscalesam** [comment]
@jjyao I think we can abandon this per @scv119 's PR right? cc @jerome-habana

**Comment by jerome-habana** [comment]
changes merged with https://github.com/ray-project/ray/issues/38504
